online-shop/src/app/app-routing.module.ts

```diff
@@ -0,0 +1,19 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { ProductListComponent } from './product-list/product-list.component';
+import {ProductDetailsComponent} from "./product-details/product-details.component";
+import {HeaderComponent} from "./header/header.component";
+import {AppComponent} from "./app.component";
+
+const routes: Routes = [
+  {path: '', redirectTo: 'home', pathMatch: 'full'},
+  {path: 'home', component: HeaderComponent},
+  {path: 'products', component: ProductListComponent},
+  {path:'products/product/:id', component: ProductDetailsComponent}
```

Comment:
Get used to always format your code. It is a good practice and your colleagues(and future you) will be thankful.
Look up to prettier/es-lint (https://dev.to/this-is-angular/configure-prettier-and-eslint-with-angular-526c)
use the default rule set of Angular for now, it should to the trick